ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -317,11 +363,22 @@ boolean isCaughtUp() {
   void shutdown() throws InterruptedException, StoreException, IOException {
     if (running.get()) {
       running.set(false);
-      synchronized (lock) {
-        lock.notify();
+      boolean isPaused = isPaused();
+      if (isPaused) {
+        reentrantLock.lock();
+        paused.set(false);
+        pauseCondition.signal();
+        reentrantLock.unlock();
+      } else {
+        synchronized (reentrantLock) {
```

Comment:
I did think about it. But then, we need to add a new method in Time interface as below
public abstract void wait(Condition c, long ms) throws InterruptedException;
so that we can advance the time in the tests.